aten/src/ATen/native/cuda/ActivationSiluKernel.cu

```diff
@@ -47,7 +47,7 @@ void silu_backward_kernel(TensorIteratorBase& iter) {
           const opmath_t dy_acc = static_cast<opmath_t>(dy);
           const opmath_t x_acc = static_cast<opmath_t>(x);
           const opmath_t s_acc =
-              opmath_t(1) / (opmath_t(1) + c10::cuda::compat::exp(-x_acc));
```

Comment:
This doe snot look correct. Why are you changing `exp` by `c10::complex`??

Reply:
Thanks for your comment! I changed it because I got these errors

```
aten/src/ATen/native/cuda/ActivationSiluKernel.cu(24): error: no instance of overloaded function "c10::cuda::compat::exp" matches the argument list
            argument types are: (c10::complex<float>)

aten/src/ATen/native/cuda/ActivationSiluKernel.cu(39): error: no instance of overloaded function "c10::cuda::compat::exp" matches the argument list
            argument types are: (c10::complex<double>)
```

Reply:
sure, but does this change make sense for real tensors?

Reply:
Ah OK.... Should I change `AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2` to `AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2`? I can't think of another solution

Reply:
No, I think that part is fine. I am solely concerned about changing and exponential by a complex number. That seems to change the semantics of the operation, and that may be why the tests are failing

Reply:
How can I solve the tests then? Do you have any ideas?

Reply:
I'd check how similar operations implement similar logic.

Reply:
@shogohida you just need to add exp for `c10::complex<{float/double}>` here: https://github.com/pytorch/pytorch/blob/55e8b5c126b3cd6fb23ee6f074651d891b68d456/c10/cuda/CUDAMathCompat.h#L32-L37

Reply:
I think adding:
```
 __MATH_FUNCTIONS_DECL__ float exp(c10::complex<float> x) { 
   return ::cexpf(x); 
 } 
 __MATH_FUNCTIONS_DECL__ double exp(c10::complex<double> x) { 
   return ::cexp(x); 
 } 
 ```
should work. You also want to retain the exp call in
[aten/src/ATen/native/cuda/ActivationSiluKernel.cu](https://github.com/pytorch/pytorch/pull/89649/files/6d28d0e4cc34f41bf8070534c26fc23e3aa3b7f1#diff-3b3853ac3ffcfe1bdff66745d060a1b5f1845626bc1a36b993f09ddefdf71ffc).

Reply:
Did you mean these changes?

https://github.com/pytorch/pytorch/pull/89649/commits/469b9af94a2765507864e1239bbffc4eda4e05fc